File: Elixir/twosum.exs

defmodule TwoSum do
  def twosum(nums, target) do
    for n1 <- nums, n2 <- nums, n1 + n2 == target, do: [n1, n2]
  end
end
#test of twosum
IO.inspect TwoSum.twosum([2, 7, 11, 15], 9)

File: Haskell/twosum.hs
import Data.Map
pmap = Prelude.map

list = [2,7,11,15]
target = 9

twoSum :: Int->[Int]->[(Int, Int)]
twoSum n x = [(i, j) | let indexed = zip x [0..],
                       let indexedmap = fromList indexed,
                       (m,i) <- indexed,
                       member (n - m) indexedmap,
                       let j = indexedmap ! (n - m),
                       i < j]

answer = twoSum target list